### ISSUE TITLE
git-monitor-3.1.1.4 does not compile

### ISSUE THREAD

**Issue by peti** [jwiegley/gitlib#55]
http://hydra.cryp.to/build/1164344/log/raw

**Comment by jwiegley** [comment]
@peti Do you open these issues manually each time my packages break?  If so, wow, much respect.

At a local Hackathon I gave maintenance of `logging` over to @aupiff, so I'll modify the upper bound in Cabal for now; otherwise I'm waiting on a new release to fix the bug you mention in your log.

**Comment by peti** [comment]
OK. I'll disable builds in Nixpkgs for the time being.

**Comment by aupiff** [comment]
@peti should be resolved with package v3.0.2 which I uploaded to hackage a few moments ago

**Comment by mdorman** [comment]
Unfortunately, it's not resolved.  Since logging decided to move away from being a monad-logger front-end, it doesn't match up with with the many places parts of gitlib/* depend on monad-logger:

```
[nix-shell:~/src/gitlib/git-monitor]$ cabal build
Building git-monitor-3.1.1.4...
Preprocessing executable 'git-monitor' for git-monitor-3.1.1.4...
[1 of 1] Compiling Main             ( Main.hs, dist/build/git-monitor/git-monitor-tmp/Main.o )

Main.hs:127:22:
    No instance for (MonadLogger IO)
      arising from a use of ‘resolveReference’
    In the expression: resolveReference sref
    In a stmt of a 'do' block:
      scr <- if optResume opts then
                 resolveReference sref
             else
                 return Nothing
    In the expression:
      do { let sref = "refs/snapshots/" <> ref;
           scr <- if optResume opts then
                      resolveReference sref
                  else
                      return Nothing;
           scr' <- maybe (resolveReference "HEAD") (return . Just) scr;
           case scr' of {
             Nothing -> errorL "Failed to lookup reference"
             Just r -> do { ... } } }
```

**Comment by aupiff** [comment]
Ah. v2 and v3 differ only in that v3 has all the `monad-logger` stuff stripped out. I don't think it's coming back :(

**Comment by jwiegley** [comment]
I'll fix gitlib not to reference `monad-logger` either.  I'd already done this in the `v4` branch, but I can backport the change to v3.

**Comment by jwiegley** [comment]
I've uploaded updates to `gitlib`, `gitlib-libgit2` and `git-monitor` which remove the need for `monad-logger`, and which are confirmed to build with the latest `logging` on my machine.

**Comment by jwiegley** [comment]
@peti Can you confirm `git-monitor` builds for you, and if so, re-enable it in the Nix Haskell config?
